ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java

```diff
@@ -207,10 +207,37 @@ void removeActiveBannerAd(@NonNull String placementId) {
     }
   }
 
-  void storeActiveBannerAd(@NonNull String placementId, @NonNull VungleBannerAdapter instance) {
-    if (!mVungleBanners.containsKey(placementId)) {
+  /**
+   * maintain banner adapter instance, replace banner adapter instance if banner view changed.
+   *
+   * @param placementId placement
+   * @param instance    banner adapter instance with updated banner or MREC
+   * @param updated     banner instance updated after cleanUp
+   */
+  void storeActiveBannerAd(@NonNull String placementId, @NonNull VungleBannerAdapter instance,
+      boolean updated) {
+    if (!mVungleBanners.containsKey(placementId) || updated) {
       mVungleBanners.put(placementId, instance);
       Log.d(TAG, "restoreActiveBannerAd:" + instance + "; size=" + mVungleBanners.size());
     }
   }
+
+  /**
+   * cleanup active banner and MREC instance
+   *
+   * @param placementId placement
+   */
+  void cleanUpActiveBannerAd(@NonNull String placementId) {
```

Comment:
remove, cleanup and destroy BannerAd methods need more detailed description in comments to avoid confusion what each does , as they all sound similar

Reply:
will be adding by comments in upstream PR.